android/capacitor/src/main/java/com/getcapacitor/plugin/camera/ImageUtils.java

```diff
@@ -19,24 +19,38 @@
 public class ImageUtils {
 
   /**
-   * Resize an image to the given width and height. Leave one dimension 0 to
-   * perform an aspect-ratio scale on the provided dimension.
+   * Resize an image to the given max width and max height. Constraint can be put
+   # on one dimension, or both. Resize will always preserve aspect ratio.
    * @param bitmap
    * @param width
    * @param height
    * @return a new, scaled Bitmap
    */
-  public static Bitmap resize(Bitmap bitmap, final int width, final int height) {
-    float aspect = bitmap.getWidth() / (float) bitmap.getHeight();
-    if (width > 0 && height > 0) {
-      return Bitmap.createScaledBitmap(bitmap, width, height, false);
-    } else if (width > 0) {
-      return Bitmap.createScaledBitmap(bitmap, width, (int) (width * 1/aspect), false);
-    } else if (height > 0) {
-      return Bitmap.createScaledBitmap(bitmap, (int)(height * 1/aspect), height, false);
+  public static Bitmap resize(Bitmap bitmap, final int desiredMaxWidth, final int desiredMaxHeight) {
+    int width = bitmap.getWidth();
+    int height = bitmap.getHeight();
+
+    // 0 is treated as 'no restriction'
+    int maxHeight = desiredMaxHeight == 0 ? height : desiredMaxHeight;
+    int maxWidth = desiredMaxWidth == 0 ? width : desiredMaxWidth;
+
+    // resize with preserved aspect ratio
+    int newWidth = Math.min(width, maxWidth);
+    int newHeight = (height * newWidth) / width;
+
+    if (newHeight > maxHeight) {
+      newWidth = (width * maxHeight) / height;
+      newHeight = maxHeight;
     }
 
-    return bitmap;
+    newWidth = Math.round(newWidth);
```

Comment:
They are `int` already, so round won't do anything on them.
But `int` just removes everything after the `.`.  (i.e. 2.9 becomes 2, not 3).
So you probably need to use float for the variables where you put a value that is divided, and then use round on the floats to make them int.